src/Twig/MercureExtension.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Twig;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Mercure\Authorization;
+use Symfony\Component\Mercure\HubRegistry;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class MercureExtension extends AbstractExtension
+{
+    private HubRegistry $hubRegistry;
+    private ?Authorization $authorization;
+    private ?RequestStack $requestStack;
+
+    public function __construct(HubRegistry $hubRegistry, ?Authorization $authorization = null, ?RequestStack $requestStack = null)
+    {
+        $this->hubRegistry = $hubRegistry;
+        $this->authorization = $authorization;
+        $this->requestStack = $requestStack;
+    }
+
+    public function getFunctions()
+    {
+        return [new TwigFunction('mercure', [$this, 'mercure'])];
```

Comment:
I wonder because, this function *does* something - it returns the URL, but also sets the cookie - if the name should reflect that - like `init_mercure()`. I'm not positive that's better - just a thought.

Reply:
Yes, I'm also hesitating. On one hand, `mercure_init` is more explicit, and I like it, on another hand `mercure` is shorter and makes it more clear that the helper returns the URL to use to connect.

Reply:
since both names are not fully reflecting what it does, I'd keep `mercure` for shortness.